### PR TITLE
Automate Release to docker.io #3

### PR DESCRIPTION
**Pull request by jsi**
I have created an Action that builds and publishes Docker images from branches.  Snapshots are ignored.  This has been tested in several setups to make sure the logic is good.

### CODE REVIEW

**Comment by jsi** [comment]
Remember to "Squash and Merge" ;)

File: .github/workflows/docker-main.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Set run date
+        run: echo "BUILD_DATE_TIME=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_ENV
+
+      - name: Docker Build
+        if: ${{ ! contains( env.XP_VERSION, '-SNAPSHOT') }}
+        run: docker build ${{ env.DOCKER_BUILD_EXTRA_ARGS }} --build-arg build_date=${{ env.BUILD_DATE_TIME }} --build-arg build_base_image=${{ env.BASE_IMAGE }} --build-arg build_distro_version=${{ env.XP_VERSION }} -t ${{ env.IMAGE_FULL_TAG }} .
+
+      - name: Docker publish
+        if: ${{ ! contains( env.XP_VERSION, '-SNAPSHOT') }}
+        run: |
```
Comment (rymsha) [inline .github/workflows/docker-main.yaml:40]: Have you looked at https://github.com/docker/build-push-action#push-to-multi-registries ?
It maybe a cleaner solution

File: .github/workflows/docker-main.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          env-file: .env
+
+      - uses: docker/setup-buildx-action@v1
+
+      - uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+#      # If snapshot version, don't execute this Action
```
Comment (rymsha) [inline .github/workflows/docker-main.yaml]: Yeah that doesn't work. You could remove this commented out code

Comment (rymsha) [reply .github/workflows/docker-main.yaml:37]: Strictly speaking you may FAIL the build if it contains SNAPSHOT instead.

File: .github/workflows/docker-main.yaml (hunk 1 above)
Comment (jsi) [reply .github/workflows/docker-main.yaml:40]: No, haven't tried it, but I see what you mean.  -  Not that used to Actions yet, and the current solution does work.

Comment (jsi) [reply .github/workflows/docker-main.yaml:40]: If you insist, I will look into it tomorrow.

Comment (rymsha) [reply .github/workflows/docker-main.yaml:40]: I don't ;-)

**Review by rymsha — APPROVED** [review]

File: .github/workflows/docker-main.yaml (hunk 1 above)
Comment (rymsha) [reply .github/workflows/docker-main.yaml:40]: I created a separate issue #5

**Pull request merged**